src/fprime/common/models/serialize/enum_type.py

```diff
@@ -86,12 +111,13 @@ def deserialize(self, data, offset):
         else:
             raise TypeRangeException(int_val)
 
-    @classmethod
-    def getSize(cls):
+    def getSize(self):
         """Calculates the size based on the size of an integer used to store it"""
-        return struct.calcsize(">i")
+        return struct.calcsize(
+            FPRIME_INTEGER_METADATA[self.REP_TYPE]["struct_formatter"]
+        )
 
     @classmethod
     def getMaxSize(cls):
         """Maximum size of type"""
-        return cls.getSize()  # Always the same as getSize
+        return struct.calcsize(FPRIME_INTEGER_METADATA["U64"]["struct_formatter"])
```

Comment:
To fix the unit-tests this should read: 

```
struct.calcsize(FPRIME_INTEGER_METADATA[cls.REP_TYPE]["struct_formatter"])
Comment
```

This size of a given enum is always the same, the size of the representation type.  This is part of the enum type definition, not the enum type instance.